### ISSUE TITLE
Cache doesn't clear for node resources.

### ISSUE THREAD

**Issue by jcoyne** [ActiveTriples/ActiveTriples#38]
```ruby
2.1.2 :001 > roles = RoleAssertionsResource.new
 => #<RoleAssertionsResource:0x3fdf099b3110(default)> 
2.1.2 :002 > user = UserResource.new.tap {|u| u.name = "bob"}
 => #<UserResource: @name=["bob"]> 
2.1.2 :003 > user2 = UserResource.new.tap { |u| u.name = "sally" }
 => #<UserResource: @name=["sally"]> 
2.1.2 :004 > roles[:owner] = user
 => #<UserResource: @name=["bob"]> 
2.1.2 :005 > roles[:owner]
 => [#<UserResource: @name=["bob"]>] 
2.1.2 :006 > roles[:owner] = user2
 => #<UserResource: @name=["sally"]> 
2.1.2 :007 > roles[:owner]
 => [#<UserResource: @name=["bob"]>, #<UserResource: @name=["sally"]>] 
```

**Comment by no-reply** [comment]
The culprit here seems to be `Relation#empty_property` which seems to roundly ignore the values in the example:

https://github.com/ActiveTriples/ActiveTriples/blob/develop/lib/active_triples/relation.rb#L58-L64

**Comment by no-reply** [comment]
Reopening per #142

**Comment by no-reply** [comment]
This is fixed by #159
